File: snippets/lcd_latest_inference.ts
// See https://docs.allora.network/reference/networks for the LCD URL and
// emissions namespace of each network.
const LCD_URL = "https://allora-api.testnet.allora.network";
const EMISSIONS = "emissions/v10";

async function getLatestInference(topicId: number): Promise<any> {
  const url = `${LCD_URL}/${EMISSIONS}/latest_network_inferences/${topicId}`;
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error(`LCD request failed: ${response.status} ${response.statusText}`);
  }
  return response.json();
}

async function main() {
  const data = await getLatestInference(1);
  // combined_value is a list of labeled values; a single-output topic has one entry ("y")
  console.log(`Latest inference: ${data.network_inferences.combined_value[0].value}`);
  console.log(`Inference block height: ${data.inference_block_height}`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: snippets/lcd_latest_inference.py
import json
import urllib.request

# See https://docs.allora.network/reference/networks for the LCD URL and
# emissions namespace of each network.
LCD_URL = "https://allora-api.testnet.allora.network"
EMISSIONS = "emissions/v10"


def get_latest_inference(topic_id):
    url = f"{LCD_URL}/{EMISSIONS}/latest_network_inferences/{topic_id}"
    with urllib.request.urlopen(url, timeout=30) as response:
        return json.load(response)


data = get_latest_inference(1)
# combined_value is a list of labeled values; a single-output topic has one entry ("y")
print(f"Latest inference: {data['network_inferences']['combined_value'][0]['value']}")
print(f"Inference block height: {data['inference_block_height']}")
